stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+            .build()
+    )
+
+    val mixedMediaPreviewLauncher = rememberLauncherForActivityResult(
+        contract = MediaGalleryPreviewContract(),
+        onResult = { result -> onMediaGalleryPreviewResult(result) }
+    )
+
+    // Used to refresh the request for the current page
+    // if it has previously failed.
+    if (data != null && connectionState == ConnectionState.CONNECTED &&
+        painter.state is AsyncImagePainter.State.Error
+    ) {
+        retryHash++
+    }
+
+    val areVideosEnabled = ChatTheme.videoThumbnailsEnabled
+
+    Box(
+        modifier = modifier
+            .background(Color.Black)
+            .fillMaxWidth()
+            .combinedClickable(
+                interactionSource = MutableInteractionSource(),
+                indication = rememberRipple(),
+                onClick = {
+                    mixedMediaPreviewLauncher.launch(
+                        MediaGalleryPreviewContract.Input(
+                            message = message,
+                            initialPosition = attachmentPosition,
+                            videoThumbnailsEnabled = areVideosEnabled
+                        )
+                    )
+                },
+                onLongClick = { onLongItemClick(message) }
+            ),
+        contentAlignment = Alignment.Center
+    ) {
+        val backgroundColor =
+            if (isImage) ChatTheme.colors.imageBackgroundMessageList
+            else ChatTheme.colors.videoBackgroundMessageList
+
+        Image(
+            modifier = modifier
+                .fillMaxSize()
+                .background(backgroundColor),
+            painter = painter,
+            contentDescription = null,
+            contentScale = ContentScale.Crop
+        )
+
+        MediaPreviewPlaceHolder(
+            asyncImagePainterState = painter.state,
+            progressIndicatorStrokeWidth = 3.dp,
+            progressIndicatorFillMaxSizePercentage = 0.25f,
+            isImage = isImage,
+            placeholderIconTintColor = ChatTheme.colors.disabled
+        )
+
+        if (isVideo) {
+            playButton()
+        }
+    }
+}
+
+/**
+ * A simple play button that is overlaid above
+ * video attachments.
+ *
+ * @param modifier The modifier used for styling.
+ */
+@Composable
+public fun PlayButton(
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier,
+        contentAlignment = Alignment.Center
+    ) {
+        Column {
+            Image(
+                modifier = Modifier
+                    .fillMaxSize(0.85f)
+                    .alignBy { measured ->
+                        // emulated offset as seen in the design specs,
+                        // otherwise the button is visibly off to the start of the screen
+                        -(measured.measuredWidth * 1 / 8)
+                    },
+                painter = painterResource(id = R.drawable.stream_compose_ic_play),
+                contentDescription = null,
```

Comment:
might be useful to add the description given that htis is an action

Reply:
I'll add description since I think we should pay more attention to accessibility, but as you noted elsewhere,, this is not an action but rather a common icon.